runtime/src/block_min_prioritization_fee_cache.rs

```diff
@@ -0,0 +1,595 @@
+use {
+    crate::{
+        block_min_prioritization_fee::*,
+        block_min_prioritization_fee_cache_query::BlockMinPrioritizationFeeCacheQuery,
+        block_min_prioritization_fee_cache_update::BlockMinPrioritizationFeeCacheUpdate,
+    },
+    solana_sdk::{
+        clock::Slot, pubkey::Pubkey, saturating_add_assign, transaction::SanitizedTransaction,
+    },
+    std::collections::HashMap,
+};
+
+/// The maximum number of blocks to keep in `BlockMinPrioritizationFeeCache`; States from
+/// up to 150 recent blocks should be sufficient to estimate minimal prioritization fee to
+/// land transactions to current block.
+const NUMBER_OF_RECENT_BLOCKS: usize = 150;
+
+/// Holds up to NUMBER_OF_RECENT_BLOCKS recent block's min prioritization fee for block,
+/// and for each writable accounts per block.
+pub struct BlockMinPrioritizationFeeCache {
```

Comment:
Yea, agree that `Block` is pretty redundant. But wondering if it'd be misleading it caches _all_ prioritization fee if just  `PrioritizationFeeCache`? Since it only captures the min fee, maybe still keep `Min` in the name, `Min PrioritizationFeeCache`?

Reply:
I'll make similar changes as you suggested elsewhere: removing `BlockMin` form struct name, qualify fields with `min` if appropriate.